screens/GroupInvite.tsx

```diff
@@ -100,24 +92,20 @@ export default function GroupInviteScreen({
     });
     const groupId = groupInvite.groupId;
     // Group ID is not available on previous versions of the app, so we need to fetch the groups
-    const groupsBeforeJoining = groupId
-      ? { ids: [], byId: {} }
-      : await fetchGroupsQuery(account);
+    const groupsBeforeJoining = await fetchGroupsQuery(account);
+    if (groupId && groupsBeforeJoining.byId[getTopicFromGroupId(groupId)]) {
+      // User has already been added to the group
+      handleNewGroup(groupsBeforeJoining.byId[getTopicFromGroupId(groupId)]);
+      return;
+    }
     logger.debug(
       `[GroupInvite] Before joining, group count = ${groupsBeforeJoining.ids.length}`
     );
-    let joinRequestId = getInviteJoinRequest(account, groupInvite?.id);
```

Comment:
was using an old `inviteJoinRequest` perhaps the culprit for people's joins timing out do you think?

if not - why was this removed in favor of always creating one?

Reply:
2 fold:
1. If a user presses again it will actually send the notification again, so has a chance to potentially reduce the manual step
2. Moved this check to the backend relevant PR is here: https://github.com/ephemeraHQ/converse-backend/pull/39